### PR TITLE
Implement some system settings functions

### PR DESCRIPTION
**Pull request by greggameplayer**
- Implement GetAccountSettings & SetAccountSettings

- Implement GetDataDeletionSettings & SetDataDeletionSettings

- Implement GetLdnChannel & SetLdnChannel

### CODE REVIEW

File: src/core/hle/service/set/set_sys.cpp [hunk 1]
```diff
@@ -28,6 +28,63 @@ void SET_SYS::SetColorSetId(Kernel::HLERequestContext& ctx) {
     LOG_DEBUG(Service_SET, "called");
 }
 
+void SET_SYS::GetAccountSettings(Kernel::HLERequestContext& ctx) {
```
Comment (ogniK5377) [inline src/core/hle/service/set/set_sys.cpp]: This is a stub, label it as so

File: src/core/hle/service/set/set_sys.cpp [hunk 2]
```diff
@@ -28,6 +28,63 @@ void SET_SYS::SetColorSetId(Kernel::HLERequestContext& ctx) {
...
+    rb.Push(RESULT_SUCCESS);
+
+    LOG_DEBUG(Service_SET, "called");
+}
+
+void SET_SYS::GetDataDeletionSettings(Kernel::HLERequestContext& ctx) {
+    IPC::ResponseBuilder rb{ctx, 3};
+
+    rb.Push(RESULT_SUCCESS);
+    rb.Push<u64>(data_deletion_settings);
```
Comment (ogniK5377) [inline src/core/hle/service/set/set_sys.cpp]: This is suppose to be 2 u32s, not a single u64.

File: src/core/hle/service/set/set_sys.cpp (hunk 1 above)
Comment (ogniK5377) [inline src/core/hle/service/set/set_sys.cpp]: I believe the default value is read from a file on the system here

File: src/core/hle/service/set/set_sys.cpp [hunk 3]
```diff
@@ -28,6 +28,63 @@ void SET_SYS::SetColorSetId(Kernel::HLERequestContext& ctx) {
...
+    IPC::RequestParser rp{ctx};
+    data_deletion_settings = rp.Pop<u64>();
+
+    IPC::ResponseBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
+
+    LOG_DEBUG(Service_SET, "called");
+}
+
+void SET_SYS::GetLdnChannel(Kernel::HLERequestContext& ctx) {
```
Comment (ogniK5377) [inline src/core/hle/service/set/set_sys.cpp]: Same as GetAccountSettings

File: src/core/hle/service/set/set_sys.cpp [hunk 4]
```diff
@@ -28,6 +28,63 @@ void SET_SYS::SetColorSetId(Kernel::HLERequestContext& ctx) {
...
+    IPC::RequestParser rp{ctx};
+    account_settings = rp.Pop<u32>();
+
+    IPC::ResponseBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
+
+    LOG_DEBUG(Service_SET, "called");
+}
+
+void SET_SYS::GetDataDeletionSettings(Kernel::HLERequestContext& ctx) {
```
Comment (ogniK5377) [inline src/core/hle/service/set/set_sys.cpp]: Same as GetAccountSettings

**Comment by simontime** [comment]
This entire commit is flawed

**Pull request closed without merging**